runtime/parachains/src/configuration.rs

```diff
@@ -1064,13 +1064,23 @@ impl<T: Config> Pallet<T> {
 	pub(crate) fn initializer_finalize() {}
 
 	/// Called by the initializer to note that a new session has started.
-	pub(crate) fn initializer_on_new_session(session_index: &SessionIndex) {
+	///
+	/// Returns the configuration that was actual before the session change and the configuration
+	/// that became active after the session change. If there were no scheduled changes, both will
+	/// be the same.
+	pub(crate) fn initializer_on_new_session(
+		session_index: &SessionIndex,
+	) -> (HostConfiguration<T::BlockNumber>, HostConfiguration<T::BlockNumber>) {
```

Comment:
We could make errors way less likely here if we returned a struct with members: `old`, `new` or something. A tuple with two identically typed values feels quite error prone.

Reply:
Fair, will do.